File: application/src/main/java/whenyourcar/application/facade/user/UserLikeFacade.java
package whenyourcar.application.facade.user;

import whenyourcar.domain.entity.Car;
import whenyourcar.domain.entity.User;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;
import whenyourcar.application.dto.user.like.UserLikeResponse;
import whenyourcar.application.service.car.CarCommonService;
import whenyourcar.application.service.user.UserCommonService;
import whenyourcar.application.service.user.UserLikeService;


@Service
@RequiredArgsConstructor
public class UserLikeFacade {
    private final UserLikeService userLikeService;
    private final UserCommonService userCommonService;
    private final CarCommonService carCommonService;

    public void postUserLike(String email, Long carId) {
        User user = userCommonService.getUserByEmail(email);
        Car car = carCommonService.getCarById(carId);
        userLikeService.postUserLike(car, user);
    }

    public UserLikeResponse.UserLikesResponseDto getUserLikes(String email, Pageable pageable)  {
        User user = userCommonService.getUserByEmail(email);
        return userLikeService.getUserLikes(user, pageable);
    }

    public void deleteUserLike(String email, Long userLikeId) {
        userLikeService.deleteUserLike(email, userLikeId);
    }
}

File: application/src/main/kotlin/howmuchcar/application/facade/user/UserLikeFacade.kt
package howmuchcar.application.facade.user

import howmuchcar.application.dto.user.UserLikeResponse
import howmuchcar.application.port.`in`.car.CarSearchUseCase
import howmuchcar.application.port.`in`.user.UserLikeUseCase
import howmuchcar.domain.entity.User
import lombok.RequiredArgsConstructor
import org.springframework.data.domain.Pageable
import org.springframework.stereotype.Service

@Service
@RequiredArgsConstructor
class UserLikeFacade (
    private val userLikeUseCase: UserLikeUseCase,
    private val carSearchUseCase: CarSearchUseCase
){
    fun postUserLike(user: User,
                     carId: Long){
        val car = carSearchUseCase.getCarById(carId)
        userLikeUseCase.postUserLike(user,car)
    }

    fun getUserLikes(pageable: Pageable,
                     user: User
    ): UserLikeResponse {
        return userLikeUseCase.getUserLikes(user, pageable)
    }

    fun deleteUserLike(user: User,
                       userLikeId: Long) {
        userLikeUseCase.deleteUserLike(user, userLikeId)
    }
}